src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -0,0 +1,178 @@
+package reposense.parser;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import reposense.exception.ParseException;
+import reposense.frontend.CliArguments;
+
+public class ArgsParserTest {
+
+    private static final String projectDirectory = System.getProperty("user.dir");
+
+    private Date getDate(int year, int month, int date) {
+        return new Calendar.Builder()
+                            .setDate(year, month, date)
+                            .setTimeOfDay(0, 0, 0)
+                            .build()
+                            .getTime();
+    }
+
+    @Test
+    public void parse_allCorrectInputs_success() throws ParseException {
+        Path configFilePath = Paths.get(projectDirectory, "sample_small.csv");
+        Path outputDirectoryPath = Paths.get(projectDirectory);
+        String input = String.format("-config %s -output %s -since 01/07/2017 -until 30/11/2017",
+                configFilePath, outputDirectoryPath);
+        String[] args = input.split(" ");
+
+        CliArguments cliArguments = ArgsParser.parse(args);
+        Assert.assertEquals(configFilePath.toAbsolutePath(), cliArguments.getConfigFilePath().toAbsolutePath());
+        Assert.assertEquals(outputDirectoryPath.toAbsolutePath(), cliArguments.getTargetFilePath().toAbsolutePath());
+
+        Date expectedSinceDate = getDate(2017, Calendar.JULY, 1);
+        Date expectedUntilDate = getDate(2017, Calendar.NOVEMBER, 30);
+        Assert.assertEquals(expectedSinceDate, cliArguments.getSinceDate().get());
+        Assert.assertEquals(expectedUntilDate, cliArguments.getUntilDate().get());
+    }
+
+    @Test
+    public void parse_configFileOnly_success() throws ParseException {
+        String configFile = Paths.get(projectDirectory, "sample_small.csv").toAbsolutePath().toString();
+        String input = String.format("-config %s", configFile);
+        String[] args = input.split(" ");
+        CliArguments cliArgumentsOfAbsolutePath =  ArgsParser.parse(args);
+
+        input = String.format("-config %s", "sample_small.csv");
+        args = input.split(" ");
+        CliArguments cliArgumentsOfRelativePath = ArgsParser.parse(args);
+
+        Assert.assertEquals(cliArgumentsOfAbsolutePath.getConfigFilePath(),
+                cliArgumentsOfRelativePath.getConfigFilePath());
```

Comment:
You shouldn't parse both to compare. Once should be thrown into parse to obtain the actual, the other should be the expected value.